src/helpers.rs

```diff
@@ -447,29 +447,77 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriEvalContextExt<'mir, 'tcx
         }
     }
 
+    fn read_os_str_from_target_str<'a>(&'a self, scalar: Scalar<Tag>) -> InterpResult<'tcx, OsString> {
+        if cfg!(target_os = "unix") {
```

Comment:
You are testing the wrong target here. This is the *target Miri is compiled for*, but what you want is the *target Miri is interpreting*. See e.g. [here](https://github.com/rust-lang/miri/blob/97c257530bd1f7c54f4e805f701942b8ed335175/src/machine.rs#L109) for how to determine that.